addons/website_slides_survey/models/slide_slide.py

```diff
@@ -66,27 +66,25 @@ def _action_set_viewed(self, target_partner, quiz_attempts_inc=False):
                     }
                 )
 
-    def _get_slide_certification_data(self):
+    def _generate_certification_url(self):
         certification_data = {}
-        if not all(slide.channel_id.is_member for slide in self):
-            for s in self:
-                user_input = s.survey_id._create_answer(
-                    partner=self.env.user.partner_id,
-                    check_attempts=False,
-                    test_entry=True, **{
-                        'slide_id': s.id
-                    }
-                )
-                certification_data[s.id] = user_input._get_survey_url()
-        else:
-            for slide in self:
-                if not self.env.user._is_public() and slide.slide_type == 'certification' and slide.survey_id:
-                    if slide.channel_id.is_member:
-                        user_membership_id_sudo = slide.user_membership_id.sudo()
-                        quizz_passed = user_membership_id_sudo.survey_quizz_passed
-                        if not quizz_passed and user_membership_id_sudo.user_input_ids:
-                            last_user_input = next(user_input for user_input in user_membership_id_sudo.user_input_ids.sorted(
-                                lambda user_input: user_input.create_date, reverse=True
-                            ))
-                            certification_data[slide.id] = last_user_input._get_survey_url()
+        for slide in self:
```

Comment:
self.filtered(lambda slide: slide.slide_type == 'certification' and slide.survey_id)